Gunshot residue tweaks and forensic fixes

comma opened the pull request:
Leaving GSR on organs (with a smol burn) when pointblanking
GSR is now stored on hand organ rather than on human as a whole.
Leaving GSR on actual gun
Leaving GSR on turf (30% chance)

Seemingly unrelated changes are a fix for a bug I discovered. Turns out 'get item covering this' proc didn't work as I believed, and expected bitflags. Adds one that takes target zone instead.

Fixes #23023
Fixes #23106
Fixes forensic samples being treated as general objects in microscope.
Not getting smell of blood until there's actual blood in the list.

MistakeNot4892 commented on code/modules/detectivework/forensics.dm:
```diff
@@ -205,8 +202,7 @@ atom/proc/add_fibers(mob/living/carbon/human/M)
 			to_chat(src, "<span class='notice'>You notice some fibers embedded in \the [A]</span>")
 		if(A.fingerprints && A.fingerprints.len > 0)
```
could make this LAZYLEN while you're in the area.

PsiOmegaDelta commented on code/modules/detectivework/tools/swabs.dm:
```diff
@@ -44,26 +44,21 @@
+			afterattack(C, user, 1) //Lazy but this would work
 			return
-		user.visible_message("[user] swabs [H]'s palm for a sample.")
+		user.visible_message("[user] swabs [H]'s [O] for a sample.")
```
This will be printed as "McDective swabs McUrist's **the** hand for a sample"
You probably want `[O.name]` or corresponding var for organs.

comma replied:
Yeah, I keep forgetting organs are annoying like that.

PsiOmegaDelta commented on code/modules/detectivework/tools/swabs.dm:
```diff
@@ -2,7 +2,7 @@
 	desc = "A sterilized cotton swab and vial used to take forensic samples."
 	icon_state = "swab"
-	var/gsr = 0
+	var/list/gsr
```
This variable name isn't very telling of what it is.

comma replied:
I mean it's an acronym, GSR, but might as well rename while I'm in the area.

PsiOmegaDelta commented on code/modules/mob/living/carbon/human/unarmed_attack.dm:
```diff
@@ -94,7 +94,7 @@ var/global/list/sparring_attack_cache = list()
 		target.apply_effect(3, WEAKEN, armour)
 
-	var/obj/item/clothing/C = target.get_covering_equipped_item(zone)
+	var/obj/item/clothing/C = target.get_covering_equipped_items_by_zone(zone)
```
`get_covering_equipped_items_by_zone()` returns a list of items unlike the previous `get_covering_equipped_item()`
This check will fail.

PsiOmegaDelta commented on code/modules/mob/living/carbon/human/inventory.dm:
```diff
@@ -424,4 +424,10 @@ This saves us from having to call add_fingerprint() any time something is put in
+/mob/living/carbon/human/proc/get_covering_equipped_items_by_zone(var/zone)
+	var/obj/item/organ/external/O = get_organ(zone)
+	if(O)
+		return get_covering_equipped_items(O.body_part)
```
`get_covering_equipped_items()` always returns a list, ensuring null-checks aren't needed.
This proc should probably follow that convention, i.e. `return list()` below the if-statement.

PsiOmegaDelta commented on code/modules/mob/mob.dm:
```diff
@@ -638,7 +638,7 @@
 			to_chat(src, "<span class='danger'>Pulling \the [H] in their current condition would probably be a bad idea.</span>")
 
-		var/obj/item/clothing/C = H.get_covering_equipped_item(BP_CHEST)
+		var/obj/item/clothing/C = H.get_covering_equipped_items_by_zone(BP_CHEST)
```
Same as above, you now get a list instead of a singular item.

PsiOmegaDelta commented on code/modules/projectiles/ammunition.dm:
```diff
@@ -37,18 +37,26 @@
+		else if(H.r_hand == G)
+			zone = BP_R_HAND
+		if(zone)
+			var/target = H.get_covering_equipped_items_by_zone(zone)
```
`put_residue_on()` doesn't appear able to handle lists of items.

PsiOmegaDelta commented on code/modules/mob/grab/grab_object.dm:
```diff
@@ -158,7 +158,7 @@
 
 /obj/item/grab/proc/leave_forensic_traces()
-	var/obj/item/clothing/C = affecting.get_covering_equipped_item(target_zone)
+	var/obj/item/clothing/C = affecting.get_covering_equipped_items_by_zone(target_zone)
```
Another case of going from singular item to list of items that won't succeed.

comma committed:
Gunshot residue tweaks and forensic fixes

Leaving GSR on organs (with a smol burn) when pointblanking
GSR is now stored on hand organ rather than on human as a whole.
Leaving GSR on actual gun
Leaving GSR on turf (30% chance)

Seemingly unrelated changes are a fix for a bug I discovered. Turns out 'get item covering this' proc didn't work as I believed, and expected bitflags. Adds one that takes target zone instead.

Fixes #23023
Fixes #23106

comma commented:
Changed get_items to get_item (using respective mob proc)

PsiOmegaDelta commented:
Cannot say I like those istype checks..
